### PR TITLE
Helper-cli: clean up also the error resolutions

### PR DESCRIPTION
**Pull request by fviernau**

### CODE REVIEW

File: helper-cli/src/main/kotlin/commands/RemoveConfigurationEntriesCommand.kt [hunk 1]
```diff
@@ -110,7 +110,7 @@ internal class RemoveConfigurationEntriesCommand : CommandWithHelp() {
             appendln("  path excludes: $removedPathExcludes")
             appendln("  scope excludes: $removedScopeExcludes")
             appendln("  rule violation resolutions: $removedRuleViolationResolutions")
-        }.let{ println(it) }
+        }.let { println(it) }
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/RemoveConfigurationEntriesCommand.kt]: Nit: This formatting fix probably belongs to another commit.

Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/RemoveConfigurationEntriesCommand.kt]: Done.

File: helper-cli/src/main/kotlin/commands/RemoveConfigurationEntriesCommand.kt [hunk 2]
```diff
@@ -27,13 +27,18 @@ import com.here.ort.helper.CommandWithHelp
 import com.here.ort.helper.common.findFilesRecursive
 import com.here.ort.helper.common.minimize
 import com.here.ort.helper.common.replacePathExcludes
+import com.here.ort.helper.common.replaceErrorResolutions
```
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/commands/RemoveConfigurationEntriesCommand.kt]: Commit message: resultion => resolution

File: helper-cli/src/main/kotlin/common/Utils.kt [hunk 3]
```diff
@@ -411,6 +412,12 @@ internal fun OrtResult.getUnresolvedRuleViolations(): List<RuleViolation> {
     }
 }
 
+/**
+ * Return a copy with the [ErrorResolution]s replaced by the given [errorResolutions].
+ */
+internal fun RepositoryConfiguration.replaceErrorResolutions(errorResolutions: List<ErrorResolution>):
+        RepositoryConfiguration = copy(resolutions = (resolutions ?: Resolutions()).copy(errors = errorResolutions))
```
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/common/Utils.kt]: The return type should be in the same line as the rest of the signature (as in the functions below).

File: model/src/main/kotlin/OrtResult.kt [hunk 4]
```diff
@@ -254,6 +254,18 @@ data class OrtResult(
...
+     * Return all analyzer errors contained in this [OrtResult].
+     */
+    @JsonIgnore
+    fun getAnalyzerErrors(): List<OrtIssue> = analyzer?.result?.errors?.values.orEmpty().flatten()
+
+    /*
+     * Return all scan errors contained in this [OrtResult].
+     */
+    @JsonIgnore
+    fun getScanErrors(): List<OrtIssue> = scanner?.results?.collectErrors()?.values.orEmpty().flatten()
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/OrtResult.kt]: It looks like you only use these functions in combination in the next commit, you could already use `collectErrors.values.flatten()` instead without adding new functions.

File: helper-cli/src/main/kotlin/commands/RemoveConfigurationEntriesCommand.kt [hunk 5]
```diff
@@ -71,32 +71,46 @@ internal class RemoveConfigurationEntriesCommand : CommandWithHelp() {
...
+            val removedPathExcludes = (repositoryConfiguration.excludes?.paths?.size ?: 0) - pathExcludes.size
+            val removedScopeExcludes = (repositoryConfiguration.excludes?.scopes?.size ?: 0) - scopeExcludes.size
+            val removedRuleViolationResolutions = (repositoryConfiguration.resolutions?.ruleViolations?.size ?: 0) -
+                    ruleViolationResolutions.size
+
+            appendln("Removed entries:")
+            appendln()
+            appendln("  path excludes: $removedPathExcludes")
+            appendln("  scope excludes: $removedScopeExcludes")
+            appendln("  rule violation resolutions: $removedRuleViolationResolutions")
```
[the remark this answers is not in the document]
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/RemoveConfigurationEntriesCommand.kt]: Done.

File: helper-cli/src/main/kotlin/common/Utils.kt (hunk 3 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/common/Utils.kt]: Done.

File: model/src/main/kotlin/OrtResult.kt (hunk 4 above)
Comment (fviernau) [reply model/src/main/kotlin/OrtResult.kt]: Done.

File: helper-cli/src/main/kotlin/commands/RemoveConfigurationEntriesCommand.kt (hunk 2 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/RemoveConfigurationEntriesCommand.kt]: Done.
